### PR TITLE
LADX: Rework dungeon item fill 

### PR DESCRIPTION
**Pull request by zig-for**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
1. Implements `get_pre_fill_items` for LADX
2. Fixes issues with filling with more exotic mixed dungeon item fill types

## How was this tested?
Generated a couple of times

### CODE REVIEW

File: worlds/ladx/__init__.py [hunk 1]
```diff
@@ -241,41 +243,102 @@ def create_items(self) -> None:
...
+        all_dungeon_items_to_fill.sort(key=priority)
+
+        # Set up state
+        all_state = self.multiworld.get_all_state(use_cache=False)
+        # Remove dungeon items we are about to put in from the state so that we don't double count
+        for item in all_dungeon_items_to_fill:
+            all_state.remove(item)
+            
+        # ...and flag items as pre-filled
+        self.pre_fill_items.extend(all_dungeon_items_to_fill)
```
Comment (Berserker66) [inline worlds/ladx/__init__.py:336]: does this do anything? Or rather is this required?
pre fill items are meant to alert AP ahead of pre fill of what will be created during pre fill. So extending this during pre fill doesn't seem correct to me.

Comment (zig-for) [reply worlds/ladx/__init__.py:334]: fixed?

Comment (Berserker66) [reply worlds/ladx/__init__.py:334]: I don't think so? The result of get_pre_fill_items wasn't meant to mutate once pre_fill is started.

Comment (zig-for) [reply worlds/ladx/__init__.py]: fixed for real - I implemented the new version of it and forgot to remove that line

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**